test/e2e/scenario/agents.scenario.ts

```diff
@@ -72,32 +73,11 @@ describe('rum', () => {
   })
 
   it('should track xhr timings', async () => {
-    await browserExecuteAsync((baseUrl, done) => {
-      let loaded = false
-      const xhr = new XMLHttpRequest()
-      xhr.addEventListener('load', () => (loaded = true))
-      xhr.open('GET', `${baseUrl}/ok`)
-      xhr.send()
-
-      const interval = setInterval(() => {
-        if (loaded) {
-          clearInterval(interval)
-          done(undefined)
-        }
-      }, 500)
-    }, browser.options.baseUrl!)
-
-    await flushEvents()
-    const timing = (await waitServerRumEvents()).find(
-      (event) =>
-        event.evt.category === 'resource' && (event as RumResourceEvent).http.url === `${browser.options.baseUrl}/ok`
-    ) as RumResourceEvent
-
-    expect(timing as any).not.toBe(undefined)
-    expect(timing.http.method).toEqual('GET')
-    expect((timing.http as any).status_code).toEqual(200)
-    expect(timing.duration).toBeGreaterThan(0)
-    expect(timing.http.performance!.download!.start).toBeGreaterThan(0)
+    const timing = await makeXHRAndCollectEvent(`${browser.config.baseUrl}/ok`)
+    expect(timing).not.toBeUndefined()
+    expect(timing!.http.method).toEqual('GET')
+    expect((timing!.http as any).status_code).toEqual(200)
+    expectToHaveValidTimings(timing!)
```

Comment:
```suggestion
    const timing = await makeXHRAndCollectEvent(`${browser.config.baseUrl}/ok`)!
    expect(timing).not.toBeUndefined()
    expect(timing.http.method).toEqual('GET')
    expect((timing.http as any).status_code).toEqual(200)
    expectToHaveValidTimings(timing)
```
and for other occurences

Reply:
Ok. In TS 3.8, I'd like to make something like:
```ts
function expectDefined<T>(value: T | undefined): asserts value is T {
  expect(value).toBeDefined()
}
```

Reply:
Does it worth the trouble?
IMO, it does not bring much more than the `!` operator and we (or at least I 😄) could forget the existence of this helper easily since it is pretty specific.

Reply:
We'll see if it can improve the situation. This is great to link the TS type to the actual JS type at runtime :) `!` should be avoided if possible, in the same way as `as` casts.

Reply:
ok, let's try